File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    let count = 0;

    for (let i = 0; i < s.length; i++) {
        if (s[i] === '(') {
            count++;
        } else if (s[i] === ')') {
            count--;
            // 만약 count가 음수가 되면 짝이 맞지 않는 것이므로 false 반환
            if (count < 0) return false;
        }
    }
    // 최종 count가 0이면 짝이 맞음
    return count === 0;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>

using namespace std;

stack<int> stk;

bool solution(string s)
{
    bool answer = false;
    
    for(int i=0;i<s.size();i++){
        if(s[i] == '(') stk.push(0);
        else if(stk.size()){
            stk.pop();
        }
        else return answer;
    }
    if(!stk.size()) answer = true;


    return answer;
}